tesseract_collision/core/include/tesseract_collision/core/types.h

```diff
@@ -460,6 +460,80 @@ struct CollisionCheckConfig
   /** @brief Secifies the mode used when collision checking program/trajectory. Default: ALL */
   CollisionCheckProgramType check_program_mode{ CollisionCheckProgramType::ALL };
 };
+
+struct ContactTrajectorySubstepResults
+{
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+  ContactTrajectorySubstepResults() = default;
+  ContactTrajectorySubstepResults(const int& substep, Eigen::VectorXd start_state, Eigen::VectorXd end_state);
```

Comment:
`int` should be passed by value.

Reply:
Eigen types need to be passed by const&.

Reply:
I updated these and added a `// NOLINT` so that it passes clang-tidy